packages/cells/src/widget.ts

```diff
@@ -1390,26 +1390,30 @@ export class CodeCell extends Cell<ICodeCellModel> {
     // to changes in metadata until we have fully committed our changes.
     // Otherwise setting one key can trigger a write to the other key to
     // maintain the synced consistency.
-    this.model.sharedModel.transact(() => {
-      super.saveCollapseState();
+    this.model.sharedModel.transact(
+      () => {
+        super.saveCollapseState();
 
-      const collapsed = this.model.getMetadata('collapsed');
+        const collapsed = this.model.getMetadata('collapsed');
 
-      if (
-        (this.outputHidden && collapsed === true) ||
-        (!this.outputHidden && collapsed === undefined)
-      ) {
-        return;
-      }
+        if (
+          (this.outputHidden && collapsed === true) ||
+          (!this.outputHidden && collapsed === undefined)
+        ) {
+          return;
+        }
 
-      // Do not set jupyter.outputs_hidden since it is redundant. See
-      // and https://github.com/jupyter/nbformat/issues/137
-      if (this.outputHidden) {
-        this.model.setMetadata('collapsed', true);
-      } else {
-        this.model.deleteMetadata('collapsed');
-      }
-    }, false);
+        // Do not set jupyter.outputs_hidden since it is redundant. See
+        // and https://github.com/jupyter/nbformat/issues/137
+        if (this.outputHidden) {
+          this.model.setMetadata('collapsed', true);
+        } else {
+          this.model.deleteMetadata('collapsed');
+        }
+      },
+      false,
+      'modeldb'
```

Comment:
I'm not a big fan of the word `modeldb` as it is not technically correct. As it is a trick to have a different behavior in the ydoc model (aka not emitting changes), something like `silent-change` sounds more appropriate for more.

Reply:
Indeed, we'll need to change it in jupyter-ydoc too (see [here](https://github.com/jupyter-server/jupyter_ydoc/blob/023afaf3a6f177be52a97e9c731e5d7aed44820a/javascript/src/ycell.ts#L663)).

Reply:
jupyter-ydoc v3.0.0a2 has `silent-change` instead of `modeldb`.